### PR TITLE
Refactoring featured art work loading view

### PR DESCRIPTION
**Pull request by AnnaBanani**
Refactoring featured art work loading view

### CODE REVIEW

File: TheMET/TheMET/Network/MetAPI.swift [hunk 1]
```diff
@@ -73,5 +73,27 @@ class MetAPI{
                 }
             }
     }
+    
+    func departments(completion: @escaping (DepartmentsResponse?) -> Void) {
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/MetAPI.swift:77]: These changes are not supposed to be in this branch

File: TheMET/TheMET/DataModels/Department.swift [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 import Foundation
 
-class Department {
+class Department: Decodable {
```
Comment (Ssimboss) [inline TheMET/TheMET/DataModels/Department.swift:10]: These changes are not supposed to be in this branch

File: TheMET/TheMET/FeaturedArtWorkLoadingView/LoadingPlaceholderView.swift [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    static func constructView() -> LoadingPlaceholderView {
+        let nib = UINib(nibName: LoadingPlaceholderView.xibFileName, bundle: nil)
+        return nib.instantiate(withOwner: nil).first as! LoadingPlaceholderView
+        }
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+    }
+    
+    func setup(configuration: LoadingPlaceholderConfiguration) {
```
Comment (Ssimboss) [inline TheMET/TheMET/FeaturedArtWorkLoadingView/LoadingPlaceholderView.swift:28]: I suggest to make this method `private`, add an argument `configuration: LoadingPlaceholderConfiguration` to `constructView()` method and execute `setup` there on the constructed object

File: TheMET/TheMET/FeaturedArtWorkLoadingView/LoadingPlaceholderView.swift [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    @IBOutlet var textLabel: UILabel!
+    @IBOutlet var imageView: UIImageView!
+    @IBOutlet var button: UIButton!
+    
+    static func constructView() -> LoadingPlaceholderView {
+        let nib = UINib(nibName: LoadingPlaceholderView.xibFileName, bundle: nil)
+        return nib.instantiate(withOwner: nil).first as! LoadingPlaceholderView
+        }
+    
+    override func awakeFromNib() {
```
Comment (Ssimboss) [inline TheMET/TheMET/FeaturedArtWorkLoadingView/LoadingPlaceholderView.swift:24]: This override does nothing, pls remove it

File: TheMET/TheMET/Network/DepartmentsResponse.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+//
+//  Created by Анна Ситникова on 16/05/2023.
+//
+
+import Foundation
+import UIKit
+
+class DepartmentsResponse: Decodable {
+    
+    let departments: [Department]
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/DepartmentsResponse.swift:13]: These changes are not supposed to be in this branch

**Review by Ssimboss — CHANGES_REQUESTED** [review]

File: TheMET/TheMET/FeaturedArtWorkLoadingView/LoadingPlaceholderView.swift (hunk 3 above)
Comment (AnnaBanani) [reply TheMET/TheMET/FeaturedArtWorkLoadingView/LoadingPlaceholderView.swift:28]: fixed

**Pull request merged**